bootloader/core/util.c

```diff
@@ -78,3 +78,43 @@ uint16_t compute_checksum(const uint8_t * begin, const uint8_t * end) {
     }
     return (~computed_checksum + 1) & 0xFFFF;
 }
+
+
+
+/**
+ * Iterate a byte buffer creating double word values and addresses.
+ *
+ * Writes to G4 and L5 flash happen in 64 bit data increments.
+ * @param address The starting flash address
+ * @param buffer A byte buffer
+ * @param length Length in bytes of the buffer.
+ * @param callback A callback that will be called for each 64 bit address
+ * @return True on success
+ */
+bool dword_address_iter(uint32_t address, const uint8_t * buffer, uint8_t length, address_iter_callback callback) {
+    if (!buffer || length <= 0) {
+        return true;
+    }
+    uint64_t double_word = 0;
```

Comment:
I was looking up what a double word is and I got a lot of conflicting reports.
Is it just a name for a unsigned 64 bit int? Kind of like longs and shorts?

Reply:
`word` is platform dependent. In the case  of this processor it is 32 bits. So a double word is 64 bits.